x-pack/plugins/security_solution/common/detection_engine/schemas/common/schemas.ts

```diff
@@ -294,6 +294,7 @@ export const toOrUndefined = t.union([to, t.undefined]);
 export type ToOrUndefined = t.TypeOf<typeof toOrUndefined>;
 
 export const type = t.keyof({
+  eql: null,
```

Comment:
does it make sense from a schema perspective to have `eql` be its own type? vs just another language for query

Reply:
@rw-access that's an interesting point. In general, a rule's type could be derived from its other fields, and we're mainly using it as a shortcut for that determination. I don't believe there's currently any ambiguity that the `type` field clarifies.

Another consideration is `signal.rule.type`, which we currently use to e.g. determine what actions can be taken on a given alert. We could compute the type when the alert is generated so as not to have to fetch the rule and derive its type, but that would introduce some inconsistency with `signal.rule` being a strict subset of the rule's fields.

Since both the querying and the alert generation are unique to EQL, I'm inclined to promote that "up" to the rule `type` as opposed to just `query`/`language`, but I'm happy to hear your thoughts.

Reply:
> Since both the querying and the alert generation are unique to EQL, I'm inclined to promote that "up" to the rule type as opposed to just query/language, but I'm happy to hear your thoughts.

Yeah, I think EQL is very special snowflake compared to Lucene/KQL, because of the complexity in its response format. Like you said, I think that it makes sense that even though the schema of the rule itself is otherwise identical to KQL/Lucene, it's processed very differently.

I wonder if that adds some more precision to what `type: query` means. To me, I now interpret that as "find the documents that match this boolean condition". I think that helps me mentally understand the difference better. 

Thanks, this helps me. It could be worthwhile to communicate some of that reasoning in the "Rule type" part of the UI. I think "Use KQL or Lucene to detect issues across indices." makes me think about cluster health. Maybe "Use KQL or Lucene to alert when documents match a condition" could help, but I'll defer to you and the UX team.

Reply:
Those are great points! I've opened a ticket to discuss our qualification of the different rule types and how we present that to the user: https://github.com/elastic/kibana/issues/77250